lib/completion.dart

```diff
@@ -39,81 +39,102 @@ class DartCompleter extends CodeCompleter {
     CancellableCompleter completer = new CancellableCompleter();
     _lastCompleter = completer;
 
-    servicesApi.complete(request).then((CompleteResponse response) {
-      if (completer.isCancelled) return;
-
-      int replaceOffset = response.replacementOffset;
-      int replaceLength = response.replacementLength;
-
-      String replacementString =  editor.document.value.substring(
-          replaceOffset, replaceOffset + replaceLength);
-
-      List<AnalysisCompletion> analysisCompletions = response.completions.map(
-          (completion) {
-        return new AnalysisCompletion(replaceOffset, replaceLength, completion);
-      }).toList();
-
-      List<Completion> completions =  analysisCompletions.map((completion) {
-        // TODO: Move to using a LabelProvider; decouple the data and rendering.
-        String displayString = completion.isMethod
-            ? '${completion.text}${completion.parameters}' : completion.text;
-        if (completion.isMethod && completion.returnType != null) {
-          displayString += ' → ${completion.returnType}';
+    if (editor.lookingForQuickFix) {
```

Comment:
Instead of storing `lookingForQuickFix` as a public field on the editor class, let's have it be passed into the CodeCompleter.complete() method as an optional named param.